docs/concurrency-adoption-guidelines.md

```diff
@@ -0,0 +1,201 @@
+# Swift Concurrency adoption guidelines for Swift Server Libraries
+
+This writeup attempts to provide a set of guidelines to follow by authors of server-side Swift libraries. Specifically a lot of the discussion here revolves around what to do about existing APIs and libraries making extensive use of Swift NIO’s `EventLoopFuture` and related types.
+
+Swift Concurrency is a multi-year effort. It is very valuable for the server community to participate in this multi-year adoption of the concurrency features, one by one, and provide feedback while doing so. As such, we should not hold off adopting concurrency features until Swift 6 as we may miss valuable opportunity to improve the concurrency model.
+
+In 2021 we saw structured concurrency and actors arrive with Swift 5.5. Now is a great time to provide APIs using those primitives. In the future we will see fully checked Swift concurrency. This will come with breaking changes. For this reason adopting the new concurrency features can be split into two phases.
+
+
+## What you can do right now
+
+### `#if` guarding code using Concurrency
+
+In order to have code using concurrency along with code not using concurrency, you may have to `#if` guard certain pieces of code. The correct way to do so is the following:
+
+```swift
+#if compiler(>=5.5) && canImport(_Concurrency)
```

Comment:
There are bunch of places in this guide which are missing `&& canImport(_Concurrency)`. There are also places where `swift(>=5.5)` instead of `compiler(>=5.5)`.

Reply:
I think it's worth adding a note about `@available` requirements too.

Reply:
Thanks done